restore/data.go

```diff
@@ -107,7 +106,10 @@ func restoreSingleTableData(queryContext context.Context, fpInfo *filepath.FileP
 		// was expected to set up
 		if backupConfig.SingleDataFile || resizeCluster {
 			agentErr := utils.CheckAgentErrorsOnSegments(globalCluster, globalFPInfo)
-			gplog.FatalOnError(agentErr)
+			if agentErr != nil {
+				gplog.Error(agentErr.Error())
+				return nil, agentErr
```

Comment:
It's the only kind of fatal error that may happens during table restore? I think the similar problem may happens otherwise. What do we do then? Will we add one more argument <some_kind_of>Err to return?

Reply:
This is a special case only for helper errors, so that the goroutine can decide what to do next.

Reply:
I simplified solution

Reply:
But you didn't answer on my question. gprestore still may hang in case of one of the go-routines fails with fatal error. Am I correct? Why not solve problem for any fatal error in go-routines?

Reply:
> gprestore still may hang in case of one of the go-routines fails with fatal error. Am I correct?

It seems so.

Reply:
I fixed it by adding cancel in recover